src/escrow_contract/src/main.nr

```diff
@@ -27,17 +31,18 @@ pub contract Escrow {
 
     #[storage]
     struct Storage<Context> {
-        owner: PrivateImmutable<AddressNote, Context>,
+        owner: PrivateImmutable<AccountNote, Context>,
     }
 
     // Creates a new instance of the escrow contract
     #[private]
     #[initializer]
-    fn constructor(owner: AztecAddress) {
-        let mut note = AddressNote::new(owner, owner);
+    fn constructor(owner: AztecAddress, secret: Field) {
+        let mut note = AccountNote::new(owner, secret);
         let note_emission = storage.owner.initialize(&mut note);
+        let sender = context.msg_sender();
         let self_address = context.this_address();
-        note_emission.emit(encode_and_encrypt_note(&mut context, self_address, self_address));
+        note_emission.emit(encode_and_encrypt_note(&mut context, self_address, sender));
```

Comment:
why are we emitting to the sender?
if the sender needs to know this contract's secret, why don't we just emit to this contract?
if there is some technical difficulty, is better to add a TODO or a NOTE than just tweak the behaviour to an easier-to-test one 🙏  tests are important but shouldn't be driving the development, it's the other way around